playbook/app/pb_kits/playbook/pb_text_input/_text_input.scss

```diff
@@ -88,6 +88,56 @@
         border-color: $error;
       }
     }
+    .text_input_wrapper_add_on {
+      .add-on {
+        &-right {
+          .add-on-card {
+            border: 1px solid red;
+            border-left: 0;
+          }
+        }
+        &-left {
+          .add-on-card {
+            border: 1px solid red;
+            border-right: 0;
+          }
+        }
+      }
+      .border {
+        &_left {
+          &_on {
+            .card-right-aligned {
+              border-left: 0;
+            }
+          }
+          &_off {
+            .card-right-aligned {
+              border-left: 0;
+            }
+            .text_input {
+              border-right: 0 !important;
```

Comment:
We generally try to avoid `!important` in our declarations. Have you tried more specific selectors/targeting to avoid this?

Reply:
@thestephenmarshall I'm not sure how to do that, but I will look into it. Could we pair later today/tomorrow if I can't figure it out?

Reply:
@thestephenmarshall Can you elaborate more when is good to use `!important`? I've noticed it's being used a lot in sass files.